### ISSUE TITLE
Hook to track asset downloads (or views)

### ISSUE THREAD

**Issue by davidjgonzalez** [adobe/asset-share-commons#959]
**Is your feature request related to a problem? Please describe.**

Its hard for me to know what assets were downloaded and by whom.

It would be nice to have a Java hook i can implement to track asset downloads (and views) in ASC. The use of the hook would be contingent on the cache-ability of assets, however since most ASC's are behind auth, the requests always go back to AEM Publish.

**Describe the solution you'd like**

I would like to have a customizable hook that allows me to implement my own code that can track (and report to some system) when an asset is downloaded. 

It would be up to me, the customizer, to ensure the hook is fast and does not impact overall asset delivery.

**Describe alternatives you've considered**

Web Analytics, however this requires significant instrumentation and is difficult to measure properly in the context of a download.  This approach could be used in tandem w/ Web analytics, since Web analytics handles cache-able requests better (in the event cacheable content is used).

**Additional context**
